### ISSUE TITLE
Memory Leak itterations

### ISSUE THREAD

**Issue by WanderWal** [Metronoid/Traficulution#6]
After so many iterations it seems as if a memory leak is created and thus the application crashes.

Most of the time this happens when you can no longer see the cars as they seemingly have been displaced out of the view. (this has not been tested).

Please try to find if some processes and functions increase in size over time, this might cause our problem.

**Comment by msnijder30** [comment]
The issue is now fixed partially, what happened was the array **entities** was saving all the entities during the entire lifespan of the application, instead of the entities of the current generation. This was causing the memory to build up.

Another issue is that the arrays in **Iterate** aren't releasing their memory correctly. I tried to fix it by setting the arrays to **null** before setting the variable equal to a new empty array. Sadly this doesn't seem to do much and memory is still building up overtime.
